### PR TITLE
Adding CI configuration for DGX-A100 at LSU

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: .jenkins/lsu/env-gcc-12-cuda-12-dgx.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+configure_extra_options+=" -DHPX_WITH_CXX_STANDARD=${CXX_STD}"
+configure_extra_options+=" -DHPX_WITH_MALLOC=system"
+configure_extra_options+=" -DHPX_WITH_FETCH_ASIO=ON"
+configure_extra_options+=" -DHPX_WITH_COMPILER_WARNINGS=ON"
+configure_extra_options+=" -DHPX_WITH_COMPILER_WARNINGS_AS_ERRORS=OFF"
+configure_extra_options+=" -DHPX_WITH_CUDA=ON"
+configure_extra_options+=" -DHPX_WITH_NETWORKING=OFF"
+configure_extra_options+=" -DHPX_WITH_DISTRIBUTED_RUNTIME=OFF"
+configure_extra_options+=" -DHPX_WITH_ASYNC_MPI=OFF"
+configure_extra_options+=" -DCMAKE_CUDA_ARCHITECTURES='52'"
```
Comment (G-071) [inline .jenkins/lsu/env-gcc-12-cuda-12-dgx.sh:25]:
```suggestion
configure_extra_options+=" -DCMAKE_CUDA_ARCHITECTURES='80'"
```
I think this should be 80 as the node only contains A100 GPUs.

**Comment by G-071** [comment]
Trying this locally on the node I get the CUDA error '''ERROR 802:system not yet initialized''' for the failing tests.

Is the nvidia fabric manager not installed on this particular node by any chance? (see [here](https://forums.developer.nvidia.com/t/error-802-system-not-yet-initialized-cuda-11-3/234955) )

**Comment by G-071** [comment]
Al rebuilt the node and the fabric manager is now installed and running.
It seems that this fixed the issues, I am now able to run all tests (including the cuda ones) successfully on the machine.

The jenkins pipeline *should* succeed once we merge master into this branch (to get the fixes from #6248 ). Ideally, we should still fix the CMAKE_CUDA_ARCHITECTURE as suggested above. @hkaiser Can you try that? I do not seem to have permission to push directly to the dgx_a100 branch.

**Comment by hkaiser** [comment]
> Al rebuilt the node and the fabric manager is now installed and running. It seems that this fixed the issues, I am now able to run all tests (including the cuda ones) successfully on the machine.
> 
> The jenkins pipeline _should_ succeed once we merge master into this branch (to get the fixes from #6248 ). Ideally, we should still fix the CMAKE_CUDA_ARCHITECTURE as suggested above. @hkaiser Can you try that? I do not seem to have permission to push directly to the dgx_a100 branch.

That's done now. Thanks!

**Comment by hkaiser** [comment]
Brilliant, this seems to work now! Thanks @akheir, @G-071!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**